vault/vault.go

```diff
@@ -62,14 +52,20 @@ func NewSessionTokenProvider(creds *credentials.Credentials, k *CredentialKeyrin
 		return nil, err
 	}
 
+	var tokenProvider mfa.TokenProvider
+	if config.MfaToken != "" {
+		tokenProvider = mfa.KnownToken{Token: config.MfaToken}
+	} else {
+		tokenProvider = mfa.GetTokenProvider(config.MfaTokenProvider)
+	}
```

Comment:
Feels like this logic might better sit in Config? e.g. `config.GetTokenProvider()`